ui/src/test/webapp/app/query/query-service-spec.js

```diff
@@ -0,0 +1,15 @@
+describe('The Query Service', function() {
+    var queryPage;
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function(_query_, _settings_, _events_) {
+        queryPage = _query_;
```

Comment:
queryPage is it's own service used to store information like the selected operation. query is the service you use to execute operations.